services/ui-src/src/components/fields/NumberField.tsx

```diff
@@ -0,0 +1,95 @@
+import React, { useState } from "react";
+import { useFormContext } from "react-hook-form";
+// components
+import { Box } from "@chakra-ui/react";
+import { TextField } from "./TextField";
+// utils
+import {
+  CustomMasks,
+  isCustomMask,
+  makeMediaQueryClasses,
+  maskValue,
+} from "utils";
+import { InputChangeEvent, AnyObject } from "types";
+
+export const NumberField = ({
+  name,
+  label,
+  placeholder,
+  mask,
+  sxOverride,
+  nested,
+  ...props
+}: Props) => {
+  const mqClasses = makeMediaQueryClasses();
+
+  const [value, setValue] = useState(props?.hydrate || "");
```

Comment:
Suggest changing to `displayValue` for clarity / to match pattern in rest of app. 

```suggestion
  const [displayValue, setDisplayValue] = useState(props?.hydrate || "");
```